### ISSUE TITLE
Harmonize vocabulary

### ISSUE THREAD

**Issue by CharlesNepote** [openfoodfacts/folksonomy_api#11]
In the API documentation we use:
* mainly "tag"/"tags":
  * "Product Tags List"
  * "Product Tag"
  * "Product Tag Delete"
  * etc.
* "k" and "v", with no explication of their meaning (=> key, value)

### Tag, key/value
In the general public, a tag is only a keyword, not a combination of a property and a value. This meaning is shared by:
* hashtags
* microformats (where the rel-tag property is _for decentralized tagging (Folksonomy)_ ([source Wikipedia](https://en.wikipedia.org/wiki/Microformat)))

The OpenStreetMap community is using the "tag" term: _A tag consists of two items, a key and a value_ ([source](https://wiki.openstreetmap.org/wiki/Tags)). But OpenStreetMap mappers are often techies who know what is key and a value. A key is probably not very instinctive for Open Food Facts contributors.

Pros:
* already use in OpenStreetMap community

Cons:
* how to create simple tags (eg. (k="tag" and v="tobedeled") if the "tag" term is already used for the whole key/value pair?

### Pair (or triple or triple tag), property, value

Pros:
* property and value are easy to understand for the general public
* these notions are used by the semantic web community
* "tag" can be a property to allow simple tags (Eg. "tag" => "to be deleted")

Cons:
* the name of the property/value pair is not very clear: a pair? a triple (like in the semantic web community)

**Comment by stephanegigandet** [comment]
I much prefer property and value. e.g. this product has the value "Green" for the property "Color". I don't see the need to introduce the notion of a pair (property + value). e.g. for the OFF API, we never say "you can set a pair of field + value", we just say that you can set the value for some field. Same here, we can read or write the value for a property.

**Comment by aadarsh-ram** [comment]
I agree with @stephanegigandet. I had a confusion regarding 'k' and 'v' when I first tried using Folksonomy Engine. Renaming it as 'property' and 'value' would be much better, as it accurately conveys it's meaning. If required, the (property+value) pair could be named as a 'couple' instead. According to me, other people would be able to instinctively understand that a 'couple' = ('property' + 'value').

@CharlesNepote @stephanegigandet, does this make any sense?

**Comment by puneeth072003** [comment]
hey i'm new to this community where shall i start

**Comment by puneeth072003** [comment]
Can i get any guidance, like what should i learn then how can i contribute in this project

**Comment by alexgarel** [comment]
Hi @puneeth072003 can you join on [slack](https://slack.openfoodfacts.org/) ? (we should limit off-topic discussion on the issue).

For this project, key knowledge is Postgresql, Python and FastAPI. You are really welcome to contribute :-)

**Comment by kirtanchandak** [comment]
Same, I too got confused with `k` `v` first time I explored `folksonomy_api`. We can rename it to `property` and `value` everywhere. 

Would love to know your thoughts @CharlesNepote @alexgarel

**Comment by alexgarel** [comment]
I think at this point we can't change the API… we have too many client depending on it.

**Comment by shradiphylleia** [comment]
maybe providing explicit definitions and examples in the documentation might help reduce this ambiguity.

**Comment by Sunhao86** [comment]
I agree with shradiphylleia's point that since the API has too many client dependencies to change, improving the documentation is the most practical solution. I can help with:

1. Creating a clear terminology glossary that explicitly explains 'k' and 'v' as representing 'property' and 'value'
2. Adding more real-world examples to the documentation, such as:
   Example 1: k="color", v="green" representing the property "color" with value "green"
   Example 2: How to create simple tags, e.g., k="tag", v="to_be_deleted"
3. Developing a brief "terminology guide" section for new users to help them quickly understand these concepts

If the team finds this helpful, I'd be happy to submit a PR with these improvements.
